runtime/domain.c

```diff
@@ -1624,6 +1624,25 @@ void caml_interrupt_all_signal_safe(void)
   }
 }
 
+static void stw_terminate_domain(caml_domain_state *domain, void *data,
+  int participating_count,
+  caml_domain_state **participating)
+{
+  if (!pthread_equal(pthread_self(), *(pthread_t *)data)) {
+    /* Ask the matching backup thread, if started, to terminate. */
+    atomic_store_release(&domain_self->backup_thread_msg, BT_TERMINATE);
+    pthread_exit(0);
+  }
```

Comment:
If I understand the code right, the "main domain" (in fact the caller of `caml_terminate_all_domains` will not terminate its own backup thread, only other domains backup threads. Is this intentional?

Reply:
Your understanding is correct. This is not intentional, I need to fix this to also require the main domain's backup thread termination.

Reply:
Actually this is not needed - the main domain will run `domain_terminate` which already takes care of its backup thread. I have added a comment to explain this.